File: progSys/exec/python/exo5.py
import os
import sys
import subprocess

while True:
    pid = os.fork()

    if pid == 0:
        fichier = input('Fichier (Entrez "stop" pour quitter): ')

        if fichier == "stop":
            os._exit(3) 
        else:      
            subprocess.run(["emacs", fichier])
            os._exit(0)
    else:
        pid_waited, status = os.wait()
        if os.WIFEXITED(status) and os.WEXITSTATUS(status) == 3:
            sys.exit(0)

File: progSys/exec/C/exo5.c
#include <sys/wait.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
int main()
{
    char* input=malloc(10);
    pid_t pid;
    int a;
    while (1)
    {
        pid=fork();
        if (pid==0)
        {
            printf("Fichier(Entrez \"stop\" pour quitter): ");
            scanf("%s",input);

            if (strcmp(input,"stop")==0)        
                exit(3);      
            else
            {
                printf("\n");
                execl("/usr/bin/emacs","emacs",input,NULL);
            }
        }
        if (pid!=0)
        {
            wait(&a);
            if (WEXITSTATUS(a)==3)
                exit(0);
        }
    }
}